### DISCUSSION TITLE
Adding node function to resolver attribute

### DISCUSSION THREAD

**Discussion by GonzaloSP** [sequelize/sequelize#15380] [Help & Questions]
I am trying to define a custom Node.js function to a Sequelize query and I can't find the most efficient way to do it. I want to add to the booksVersions a custom field that returns the value of a Node function. In this case I named the function translateTitle. What is the right way to implement it?

My resolver looks:

``` 
function translateTitle(title) {
     //custom code
    return title_in_german;
 }

Query: {
        books: async (root, args, { models }) => {
            const options = {
                include: [
                    {
                        model: models.BooksVersions,
                        as: "booksVersions",
                        attributes: ["title",translateTitle("title")]
                    }
                ]
            }
            const Books = await models.Books.findAll(options);
            return {
                nodes: Books
            }
        },```

**Comment by ephys** [comment]
If the function you use in `attributes` needs to use a column, then you can't write it in JavaScript because it'll be executed in your database. You'd need to use a Stored Procédure instead.

Alternatively, you can fetch your data and then execute your function on each returned model.

**Reply by GonzaloSP** [reply]
Yes, I was leaning toward the latter option. Is there a place where I can find how to manipulate the model that I am returning in order to preserve the sequelize format? 

Thanks
